tests/many_to_many/tests.py

```diff
@@ -555,3 +565,72 @@ def test_custom_default_manager_exists_count(self):
         self.assertEqual(
             self.p3.article_set.exists(), self.p3.article_set.all().exists()
         )
+
+
+class ManyToManyQueryTests(TestCase):
+    """
+    #29725 - Testing SQL optimization when using
+    count() and exists() functions on queryset for
+    many to many relations
```

Comment:
To add more explanation, docstrings should state the expected behavior and omit prefixes like "Testing" since all tests test things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

The ticket number is unnecessary.